### ISSUE TITLE
Failed to initialize Metal Backend

### ISSUE THREAD

**Issue by nicholasjayantylearns** [instructlab/instructlab#1138]
**Describe the bug**
I followed the get started instructions in the Get Started docs.

When I run the `ilab serve` command I am getting the following error following an incredibly long [traceback.](https://textdoc.co/jfFqmsIAUvzDTOox) 
llama_new_context_with_model: failed to initialize Metal backend
Error creating server: failed creating the server application: Failed to create llama_context

**To Reproduce**
follow Get started instructions
Type `ilab serve`

**Expected behavior**
Server starting and sharing the appropriate info.

**Screenshots**

**Device Info (please complete the following information):**
 - Hardware Specs: Apple M1
 - OS Version: Sonoma 14.4
 - Python Version: Python 3.9.13
 - InstructLab Version: 0.14.1

**Additional context**
Not sure how to initialize the Metal backend. I've seen the following info, but none have worked:
https://github.com/ggerganov/llama.cpp/issues/5376
https://github.com/abetlen/llama-cpp-python/blob/main/docs/install/macos.md
https://github.com/abetlen/llama-cpp-python/issues/1285

**Comment by bjhargrave** [comment]
Have you run `xcode-select --install` to install the latest Xcode command line tools?

Then try 

```shell
pip cache remove llama_cpp_python
pip install --force-reinstall llama_cpp_python==0.2.55 -C cmake.args="-DLLAMA_METAL=on;-DCMAKE_OSX_ARCHITECTURES=arm64;-DCMAKE_APPLE_SILICON_PROCESSOR=arm64"
```

to rebuild llama_cpp_python.

**Comment by nicholasjayantylearns** [comment]
Let me try it and report back. Thank you!

**Comment by nicholasjayantylearns** [comment]
Downloaded the new xcode and installed the command line tools. 

After running the commands you posted it rebuilt successfully.

When I went to run ilab serve it throws an exception handling error message related to a missing llama file and something related to incompatible architectures and missing files.

 [See full error message](https://textdoc.co/pWhm7sOcIFCBNLgQ)

I looked at the get started docs again and there was a note about possible files being missing if I had other env managers installed like conda.  I then checked to see if I had any other env managers installed and sure enough, I had three separate conda envs.  I ended up removing them and the original ilab project and started over. 

After I downloaded the model again, I ran the command you shared and then `ilab serve`. It worked and I'm off and running on exploring how to contribute.  

Thank you!
